src/sentry/utils/retries.py

```diff
@@ -35,15 +35,16 @@ def __init__(self, timeout, delay=None, exceptions=(Exception,)):
         self.timeout = timeout
         self.delay = delay
         self.exceptions = exceptions
+        self.clock = time
```

Comment:
I didn't add this to the constructor since this is an attribute that should only be modified for test isolation, and didn't want to give the impression that it was something you should be using for any other reason.